apps/src/javalab/JavalabFileManagement.js

```diff
@@ -27,24 +25,31 @@ function renameProjectFile(filename, newFilename) {
   );
 }
 
-function onSave() {
-  header.showProjectSaving();
-  const filename = getStore().getState().javalab.filename;
-  const editorText = getStore().getState().javalab.editorText;
+function onSave(success, failure) {
   // TODO: enable multi-file
-  filesApi.putFile(
-    filename,
-    editorText,
-    () => {
-      // at this point project.filesVersionId will be updated and we can save
-      // sources as well.
-      project.save();
-    },
-    () => {
-      console.warn(`Javalab: error file ${filename} not saved`);
-      project.showSaveError_();
+  const filesChanged = getStore().getState().javalab.filesChanged;
```

Comment:
A common pattern I've seen when there are multiple accesses to the redux store is to assign it to a variable rather than accessing it multiple times. 
i.e. 
```
const reduxStore = getStore();
const javalabState = reduxStore.getState().javalab;
const filesChanged = javalabState.filesChanged;
```
If you do assign the state to a variable, you'll need to refresh it after dispatches complete (i.e. call `getState()` again) before accessing the state again (not an issue on this section of code).